2.4/root/usr/share/container-scripts/mongodb/common.sh

```diff
@@ -117,23 +117,60 @@ function mongo_initiate() {
 
 # get the address of the current primary member
 function mongo_primary_member_addr() {
-  local current_endpoints=$(endpoints)
-  local mongo_node="$(echo "${current_endpoints}" | grep -v "$(container_addr)" | head -1):${CONTAINER_PORT}"
-  echo -n $(mongo admin -u admin -p "${MONGODB_ADMIN_PASSWORD}" --host ${mongo_node} --quiet --eval "print(rs.isMaster().primary);")
+  endpoints | grep -v "$(container_addr)" | while read mongo_node; do
+    cmd_output="$(mongo admin -u admin -p "$MONGODB_ADMIN_PASSWORD" --host "$mongo_node:$CONTAINER_PORT" --eval 'print(rs.isMaster().primary)' --quiet || true)"
+
+    # Trying to find IP:PORT in output and filter out error message because mongo prints it to stdout
+    ip_and_port_regexp='[0-9]\+\.[0-9]\+\.[0-9]\+\.[0-9]\+:[0-9]\+'
+    if addr="$(echo "$cmd_output" | grep -x "$ip_and_port_regexp")"; then
+      echo "$addr"
+      exit 0
+    fi
+
+    echo >&2 "Cannot get address of primary from $mongo_node node: $cmd_output"
```

Comment:
I could show you how it looks like:
```console
$ mongo admin -u admin -p $MONGODB_ADMIN_PASSWORD --host localhost:21071 --eval 'print(rs.isMaster().primary)' --quiet
Thu Mar  3 13:29:09.090 Error: couldn't connect to server localhost:21071 at src/mongo/shell/mongo.js:147
exception: connect failed
$ mongo admin -u admin -p $MONGODB_ADMIN_PASSWORD --host localhost:21071 --eval 'print(rs.isMaster().primary)' --quiet >/dev/null
exception: connect failed
$ mongo admin -u admin -p $MONGODB_ADMIN_PASSWORD --host localhost:21071 --eval 'print(rs.isMaster().primary)' --quiet 2>/dev/null
Thu Mar  3 13:29:15.890 Error: couldn't connect to server localhost:21071 at src/mongo/shell/mongo.js:147
$ mongo admin -u admin -p $MONGODB_ADMIN_PASSWORD --host rs0/localhost --eval 'print(rs.isMaster().primary)' --quiet
Thu Mar  3 13:34:46.884 starting new replica set monitor for replica set rs0 with seed of localhost:27017
Thu Mar  3 13:34:46.884 successfully connected to seed localhost:27017 for replica set rs0
Thu Mar  3 13:34:46.885 changing hosts to { 0: "172.17.0.6:27017", 1: "172.17.0.5:27017", 2: "172.17.0.4:27017" } from rs0/
Thu Mar  3 13:34:46.885 trying to add new host 172.17.0.4:27017 to replica set rs0
Thu Mar  3 13:34:46.886 successfully connected to new host 172.17.0.4:27017 in replica set rs0
Thu Mar  3 13:34:46.886 trying to add new host 172.17.0.5:27017 to replica set rs0
Thu Mar  3 13:34:46.886 successfully connected to new host 172.17.0.5:27017 in replica set rs0
Thu Mar  3 13:34:46.886 trying to add new host 172.17.0.6:27017 to replica set rs0
Thu Mar  3 13:34:46.887 successfully connected to new host 172.17.0.6:27017 in replica set rs0
Thu Mar  3 13:34:46.890 Primary for replica set rs0 changed to 172.17.0.4:27017
Thu Mar  3 13:34:46.899 replica set monitor for replica set rs0 started, address is rs0/172.17.0.4:27017,172.17.0.5:27017,172.17.0.6:27017
Thu Mar  3 13:34:46.899 [ReplicaSetMonitorWatcher] starting
172.17.0.4:27017
```
As you can see Mongo always prints logs to stdout as output :-( And `--quiet` doesn't hide it.